### PR TITLE
doc(all): add technical details to documentation (Part - I)

### PR DESCRIPTION
**Pull request by leungkinghin-ct**
- Add overview section with architecture diagram in README.md
- List endpoints for following operations
   - create payment session
   - create payment
  
p.s. Remaining operations API would be included in the next pull request

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -58,9 +99,6 @@ deployAs:
         - key: CTP_PROJECT_KEY
           description: Commercetools project key
           required: true
```
Comment (praveenkumarct) [inline README.md:101]: You missed to update line number 91 and 92, Please check and correct the module names in this file.
It still says, `used by full ingestion and incremental updater modules`

File: README.md [hunk 2]
```diff
@@ -1,13 +1,55 @@
...
+2. commercetools Checkout retrieves SDK as static assets from enabler in connector.
+3. After downloading the SDK, commercetools Checkout sends request via the SDK to endpoints exposed by processor for various payment operations.
+4. The processor transforms the received request and send them to Ayden platform.
+5. Adyen processes the payment operations and return the response to processor.
+6. The processor handles the payment persistence to composable commerce, and return the result to front-end.
+7. commercetools Checkout handles the order management in composable commerce once it receives the result from the processor. 
+
 ## Prerequisite
 #### 1. commercetools composable commerce API client
 Users are expected to create API client responsible for payment management in composable commerce project. Details of the API client are taken as input as environment variables/ configuration for connect such as `CTP_PROJECT_KEY` , `CTP_CLIENT_ID`, `CTP_CLIENT_SECRET`, `CTP_REGION`. For details, please read [Deployment Configuration](./README.md#deployment-configuration).
```
Comment (praveenkumarct) [inline README.md:55]: Can we remove `CTP_REGION` from here as well?

Comment (leungkinghin-ct) [reply README.md]: https://github.com/commercetools/connect-payment-integration-adyen/pull/76/commits/491229dfad08babb346fe79a5bad76c9d08f9a73

File: README.md (hunk 1 above)
Comment (leungkinghin-ct) [reply README.md:101]: good catch
https://github.com/commercetools/connect-payment-integration-adyen/pull/76/commits/491229dfad08babb346fe79a5bad76c9d08f9a73

**Review by praveenkumarct — APPROVED** [review]

File: docs/connect-adyen.png
Comment (ddeliziact) [inline docs/connect-adyen.png:1]: This file is proably not needed as it is being rendered by github.

Comment (leungkinghin-ct) [reply docs/connect-adyen.png:1]: https://github.com/commercetools/connect-payment-integration-adyen/pull/76/commits/6feb0ec90967289eecb33c8a4f8deae5ce2c49a9

File: README.md [hunk 3]
```diff
@@ -21,20 +63,19 @@ Their values are taken as input as environment variables/ configuration for conn
...
+## Development Guide
+Regarding the development of enabler module, please refer to the following documentations:
+- [Development of Enabler](./enabler/README.md)
 
 Regarding the development of processor module, please refer to the following documentations:
 - [Development of Processor](./processor/README.md)
 
-#### 1. Register as connector in commercetools Connect
+#### Register as connector in commercetools Connect
 Follow guidelines [here](https://docs.commercetools.com/connect/getting-started) to register the connector for public/private use.
```
Comment (prateek-ct) [inline README.md:74]: Can this changed to 
"Use public connector listed in connect marketplace. If any customization done, follow guidelines [here](https://docs.commercetools.com/connect/getting-started) to register the connector for private use."

and title to "Connector in commercetools Connect"

**Review by dasanorct — APPROVED** [review]

File: README.md (hunk 3 above)
Comment (leungkinghin-ct) [reply README.md]: https://github.com/commercetools/connect-payment-integration-adyen/pull/76/commits/beedf49b43c98849b70101b1995e56aab1514931

Comment (abreu-ct) [reply README.md]: @prateek-ct missed this issue sorry

**Pull request merged**